src/lib/operators/aggregate.cpp

```diff
@@ -578,8 +583,8 @@ typename std::enable_if<func == AggregateFunction::Count, void>::type _write_agg
 // COUNT(DISTINCT) writes the number of distinct values
 template <typename ColumnType, typename AggregateType, AggregateFunction func>
 typename std::enable_if<func == AggregateFunction::CountDistinct, void>::type _write_aggregate_values(
-    std::shared_ptr<ValueColumn<AggregateType>> column,
-    std::shared_ptr<std::map<AggregateKey, AggregateResult<AggregateType, ColumnType>>> results) {
+    const std::shared_ptr<ValueColumn<AggregateType>>& column,
+    const std::shared_ptr<std::map<AggregateKey, AggregateResult<AggregateType, ColumnType>>>& results) {
```

Comment:
hijacking this to talk about what @mjendruk mentioned about `std::shared_ptr<std::vector>`:

In this particular case of `_keys_per_chunk` in Aggregate, this is a remnant of the old Aggregate code that more heavily used the visitor pattern. Before we had iterables, i think we heavily used `std::shared_ptr<*>` for visitor contexts to pass around. So as we progress more and more from visitors to iterables, we should keep in mind that changing these structures to regular vectors or maps (without a shared pointer around it) is usually a good idea. Good point!